src/DynamoApplications/StartupUtils.cs

```diff
@@ -0,0 +1,302 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Interfaces;
+using DynamoShapeManager;
+using System.Reflection;
+using System.IO;
+using Dynamo.Models;
+using Dynamo.UpdateManager;
+using Microsoft.Win32;
+using System.Diagnostics;
+using System.Threading;
+using System.Globalization;
+using NDesk.Options;
+
+namespace Dynamo.Applications.StartupUtils
+{
+    internal class PathResolver : IPathResolver
+    {
+        private readonly List<string> additionalResolutionPaths;
+        private readonly List<string> additionalNodeDirectories;
+        private readonly List<string> preloadedLibraryPaths;
+
+        public PathResolver(string preloaderLocation, IEnumerable<string> librariesToPreload)
```

Comment:
Be careful! the `IPathResolver` interface was carefully designed to to allow your implementation to encapsulate all that was needed to define a `PathResolver`. With your strategy, we now have a couple of static lists with all the libaries to load in them called things like `SandboxLibraries` and `CLILibraries`. For someone trying to use your utility methods, understanding that you have to provide one of these lists (and knowing which one to use) is actually more difficult then just having to provide the correct `PathResolver` object.

I recommend keeping the preloaded libraries lists inside the `PathResolver` implementation and making two implementations:

`SandboxPathResolver`
`CLIPathResolver`

`PathResolver`s aren't that big. You can put them in another file called `PathResolvers.cs`